app/components/primer/octicon_component.rb

```diff
@@ -27,20 +29,25 @@ class OcticonComponent < Primer::Component
     # @param size [Symbol] <%= one_of(Primer::OcticonComponent::SIZE_MAPPINGS) %>
     # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
     def initialize(icon_name = nil, icon: nil, size: SIZE_DEFAULT, **system_arguments)
-      @icon = icon_name || icon
       @system_arguments = system_arguments
-
-      @system_arguments[:class] = Primer::Classify.call(**@system_arguments)[:class]
-      @system_arguments[:height] ||= SIZE_MAPPINGS[size]
+      @system_arguments[:tag] = :svg
 
       # Filter out classify options to prevent them from becoming invalid html attributes.
       # Note height and width are both classify options and valid html attributes.
-      octicon_helper_options = @system_arguments.slice(:height, :width)
-      @system_arguments = add_test_selector(@system_arguments).except(*Primer::Classify::VALID_KEYS, :classes).merge(octicon_helper_options)
+      octicon_options = {
+        height: SIZE_MAPPINGS[fetch_or_fallback(SIZE_OPTIONS, size, SIZE_DEFAULT)]
+      }.merge(@system_arguments.slice(:height, :width, :class))
```

Comment:
this feels a bit weird to me, we are setting a specific `height` according to `size` but letting the `height` system_argument override it so I think we either want

```rb
@system_arguments[:height] = SIZE_MAPPINGS[fetch_or_fallback(SIZE_OPTIONS, size, SIZE_DEFAULT)]
octicon_options = @system_arguments.slice(:height, :width, :class)
```

or

```rb
octicon_options = {
  height: SIZE_MAPPINGS[fetch_or_fallback(SIZE_OPTIONS, size, SIZE_DEFAULT)]
}.merge(@system_arguments.slice(:width, :class))
```

also, I don't think we ever use the `class` key, it should probably be `classes` when adding custom-classes

Reply:
We have a specific, custom height and width test, which I don't think we should have allowed, but removing that functionality made me worry there would be migrations on dotcom I would need to verify. So I wrote it in such a way that height passed in will override size.